umm-spec-lib/src/cmr/umm_spec/umm_g/data_granule.clj

```diff
@@ -0,0 +1,35 @@
+(ns cmr.umm-spec.umm-g.data-granule
+  "Contains functions for parsing UMM-G JSON DataGranule into umm-lib granule model
+   DataGranule and generating UMM-G JSON DataGranule from umm-lib granule model DataGranule."
+  (:require
+   [cmr.umm.umm-granule :as g])
+  (:import cmr.umm.umm_granule.UmmGranule))
+
+(defn umm-g-data-granule->DataGranule
+  "Returns the umm-lib granule model DataGranule from the given UMM-G DataGranule."
+  [data-granule]
+  (when data-granule
+    (g/map->DataGranule
+      (let [{:keys [Identifiers DayNightFlag ProductionDateTime
+                    ArchiveAndDistributionInformation]} data-granule]
+        {:day-night DayNightFlag
+         :producer-gran-id (->> Identifiers
+                                (filter #(= "ProducerGranuleId" (:IdentifierType %)))
```

Comment:
It is probably better to rewrite this part with `some` rather than `filter` then `first`, as `some` returns right after a match is found.